imageBuilder.go

```diff
@@ -3,20 +3,67 @@ package main
 import (
 	"assignment-exec/image-builder/builder"
 	"assignment-exec/image-builder/configurations"
+	"assignment-exec/image-builder/environment"
+	"flag"
 	"log"
+	"os"
 )
 
+var pushImage = flag.Bool("pushImage", false, "Push image to docker hub")
+
+func init() {
+	flag.BoolVar(pushImage, "p", false, "Push image to docker hub")
+}
+
 func main() {
+	flag.Parse()
 	log.Println("Creating Dockerfile...")
 
+	// Setting environment variables
+	err := os.Setenv("CODE_RUNNER_YAML","code-runner.yaml")
+	if err != nil {
+		log.Fatalf("error while setting environment variables: %v", err)
+		return
+	}
+	os.Setenv("DOCKER_AUTH_YAML", "docker-auth.yaml")
+	if err != nil {
+		log.Fatalf("error while setting environment variables: %v", err)
+		return
+	}
+	os.Setenv("DOCKERFILE_PATH", "Dockerfile")
+	if err != nil {
+		log.Fatalf("error while setting environment variables: %v", err)
+		return
+	}
+	os.Setenv("DOCKER_IO_PATH", "docker.io")
+	if err != nil {
+		log.Fatalf("error while setting environment variables: %v", err)
+		return
+	}
+
 	// Unmarshal the yaml configuration file and generate a dockerfile.
-	err := configurations.WriteDockerfile()
+	err = configurations.WriteDockerfile()
 	if err != nil {
 		log.Fatalf("error while writing dockerfile: %v", err)
+		return
+	}
+
+	authData, err := builder.GetAuthData(os.Getenv(environment.DockerAuthYaml))
+	if err != nil {
+		log.Fatalf("error while reading authetication details: %v", err)
+		return
 	}
 
-	err = builder.BuildImage()
+	err = builder.BuildImage(*authData)
```

Comment:
I see that `authData` is passed for all functions in builder module.
It might be better to encapsulate all this into an `ImageBuilder` struct as shown below.
```go
type ImageBuilder struct {
    authData <type>
    // Other required fields.
}

func NewImageBuilder(...) *ImageBuilder { ... }

func (b ImageBuider) BuildImage() error { ... }

func (b ImageBuilder) PublishImage() error { ... }

Reply:
Note that we also don't need to name the function PushImage**ToHub**(...).
Just keep this in mind. I'm going to raise an issue to refactor all variable/function/struct names to be conceptual only.

Reply:
Okay